### PR TITLE
Bugfixes for basement adjacent surfaces

### PR DESCRIPTION
**Pull request by yzhou601**
## Pull Request Description

There're some bugs in eliminating basement interior surface solar absorptance:
1. Subsurfaces in conditioned basement (window modeled as door as current workaround) are not modified
2. Those basement surfaces with adjacent surfaces are not handled perfectly. The properties of adjacent surfaces are not modified correspondingly, while OpenStudio expects these two adjacent surfaces are of symmetric construction.
3. For those surfaces exterior adjacent to conditioned basement, their adjacent surfaces are supposed to be put into conditioned basement surface list instead of the surfaces themselves.

## Checklist

Not all may apply:

- [ ] EPvalidator.xml has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI

### CODE REVIEW

File: HPXMLtoOpenStudio/measure.rb [hunk 1]
```diff
@@ -612,6 +631,12 @@ def self.update_solar_absorptances(runner, model)
       innermost_material = layered_const.layers[layered_const.numLayers() - 1].to_StandardOpaqueMaterial.get
       innermost_material.setSolarAbsorptance(0.0)
       innermost_material.setVisibleAbsorptance(0.0)
+      next unless not adj_surface.nil?
+      # Create new construction in case of shared construciton.
+      layered_const_adj = OpenStudio::Model::Construction.new(model)
+      layered_const_adj.setName(cond_bsmnt_surface.construction.get.name.get + ' Reversed Bsmnt')
+      adj_surface.setConstruction(layered_const_adj)
+      layered_const_adj.setLayers(cond_bsmnt_surface.construction.get.to_LayeredConstruction.get.layers.reverse())
```
Comment (yzhou601) [inline HPXMLtoOpenStudio/measure.rb:639]: I modified the adjacent surface construction based on the new surface construction, the FT errors are gone. But not sure whether this makes sense in terms of modeling.

File: workflow/tests/hpxml_translator_test.rb [hunk 2]
```diff
@@ -672,6 +672,7 @@ def _verify_outputs(runner, rundir, hpxml_path, results)
       num_expected_kiva_instances = { 'base-foundation-ambient.xml' => 0,                # no foundation in contact w/ ground
                                       'base-foundation-multiple.xml' => 2,               # additional instance for 2nd foundation type
                                       'base-enclosure-2stories-garage.xml' => 2,         # additional instance for garage
+                                      'base-foundation-basement-garage.xml' => 2,        # additional instance for garage
```
Comment (yzhou601) [inline workflow/tests/hpxml_translator_test.rb:675]: Haha just have done the same thing with an error to push. 😛

File: HPXMLtoOpenStudio/measure.rb (hunk 1 above)
Comment (shorowit) [reply HPXMLtoOpenStudio/measure.rb:641]: It seems reasonable to me.

**Comment by shorowit** [comment]
Ran base-foundation-basement-garage.xml and the original test file in this PR and master, the FT errors were gone and simulation diffs were very tiny (0.01 MBtu at most). No diffs to any other files on the CI. Thanks @yzhou601!

**Pull request merged**